actors/migration/top.go

```diff
@@ -207,7 +239,21 @@ func MigrateStateTree(ctx context.Context, store cbor.IpldStore, stateRootIn cid
 	}
 
 	// Track deductions to burntFunds actor's balance
-	if err := p.flush(ctx, actorsIn, actorsOut); err != nil {
+	transferCancel()
+	transferFromBurnt := <-transferOutCh
+	burntFundsActor, found, err := actorsOut.GetActor(builtin.BurntFundsActorAddr)
+	if err != nil {
+		return cid.Undef, err
+	}
+	if !found {
+		return cid.Undef, xerrors.Errorf("burnt funds actor not in tree")
+	}
+	burntFundsActor.Balance = big.Sub(burntFundsActor.Balance, transferFromBurnt)
+	if burntFundsActor.Balance.LessThan(big.Zero()) {
+		return cid.Undef, xerrors.Errorf("miner transfers send burnt funds actor balance below zero")
+	}
+	err = actorsOut.SetActor(builtin.BurntFundsActorAddr, burntFundsActor)
+	if err != nil {
 		return cid.Undef, err
 	}
 
```

Comment:
Might need to reset semaphore since its a global var.  Ideally we'd move from global to some local scope but that might make code less clean with lots of closures.  Passing sync objects in functions has burned me before, maybe it s ok this time?

Not using a semaphore would also solve this problem :)